### ISSUE TITLE
for what period server crt creates by the script ?

### ISSUE THREAD

**Issue by emoxam** [Angristan/OpenVPN-install#1163]
for what period server crt creates by the script ?

I see strings
    ./easyrsa --batch build-server-full "$SERVER_NAME" nopass
    EASYRSA_CRL_DAYS=3650 ./easyrsa gen-crl
inside the script
But my server.crt is only for two years

 cat server_JsChip6h5U7Y4Je4.crt | grep -A3 Validity
        Validity
            Not Before: Jul 14 06:52:59 2023 GMT
            Not After : Oct 16 06:52:59 2025 GMT

Why is it so ?

**Comment by emoxam** [comment]
openssl x509 -in server_JsChip6h5U7Y4Je4.crt -noout -dates
notBefore=Jul 14 06:52:59 2023 GMT
notAfter=Oct 16 06:52:59 2025 GMT

**Comment by emoxam** [comment]
I found at script
    ./easyrsa --batch build-server-full "$SERVER_NAME" nopass
    EASYRSA_CRL_DAYS=365 ./easyrsa gen-crl

and add before that this string
export EASYRSA_CERT_EXPIRE=3650
After uninstall and install server cert created for 10 years.

**Comment by martijnED** [comment]
this is a really big bug, creating client certs for 10 years but not for server :/

**Comment by sabbath88** [comment]
yep. I've just run into a issue because my server cert expired. all rest certs are still valid ;)

**Comment by sabbath88** [comment]
workaround: https://github.com/angristan/openvpn-install/issues/1002#issuecomment-1229525488

**Comment by emoxam** [comment]
> workaround: [#1002 (comment)](https://github.com/angristan/openvpn-install/issues/1002#issuecomment-1229525488)
Then we need to name new client certs?

**Comment by sabbath88** [comment]
> > workaround: [#1002 (comment)](https://github.com/angristan/openvpn-install/issues/1002#issuecomment-1229525488)
> 
> Then we need to name new client certs?
> 
> 

No. If you follow strict to instructions and use the same names for server crt and key then on client side no change need to be perform.

**Comment by emoxam** [comment]
> > > workaround: [#1002 (comment)](https://github.com/angristan/openvpn-install/issues/1002#issuecomment-1229525488)
> > 
> > 
> > Then we need to name new client certs?
> 
> No. Id you followed strict to instructions and use the same names for server crt and key then on client side no change need to be performed

Wow. All clients will continue to work without any actions??

**Comment by sabbath88** [comment]
> > > > workaround: [#1002 (comment)](https://github.com/angristan/openvpn-install/issues/1002#issuecomment-1229525488)
> 
> > > 
> 
> > > 
> 
> > > Then we need to name new client certs?
> 
> > 
> 
> > No. Id you followed strict to instructions and use the same names for server crt and key then on client side no change need to be performed
> 
> 
> 
> Wow. All clients will continue to work without any actions??

Yeah, exactly (:

**Comment by emoxam** [comment]
> > > > > workaround: [#1002 (comment)](https://github.com/angristan/openvpn-install/issues/1002#issuecomment-1229525488)
> > 
> > 
> > > >
> > 
> > 
> > > >
> > 
> > 
> > > > Then we need to name new client certs?
> > 
> > 
> > > 
> > 
> > 
> > > No. Id you followed strict to instructions and use the same names for server crt and key then on client side no change need to be performed
> > 
> > 
> > Wow. All clients will continue to work without any actions??
> 
> Yeah, exactly (:

That's great! Thanks!

**Comment by angristan** [comment]
This is fixed now, default has been 10 years for server and client with a189535563f1b48605115643585e036cd5631ac1 and renewal is supported with https://github.com/angristan/openvpn-install/pull/1328
